### DISCUSSION TITLE
In a static website, what happens to environment variables?

### DISCUSSION THREAD

**Discussion by gbrandt** [flet-dev/flet#1115] [Q&A]
### Question

My app uses environment variables to load a number of items, from database location to secrets.  When I make a standalone app what happens to the env vars?  Are they dropped, baked into the code sent to the browser?

Any insight would be helpful.

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment]
When the app is published as a standalone static website environment variables cannot be used anymore. You have to be especially careful with secrets as the entire app is delivered to a user's browser and run there.

As Flet app becomes a typical SPA I guess the regular approach would work there: doing all the secret work in backend API server with cookie- or JWT-based authentication. You can use `fetch` function in Pyodide to call an API: https://pyodide.org/en/stable/usage/api/python-api/http.html
